### ISSUE TITLE
Sync issues if using rrgit in a git repo on multiple computers

### ISSUE THREAD

**Issue by AndyEveritt** [adammhaile/rrgit#2]
- Make a git repo.
- Use rrgit to clone duet config files into git repo
- Push to github
- Clone git repo elsewhere
- rrgit sync now think all the files have been changed (I think because it uses the last updated date of the file?)

Not sure on a work around for this

**Comment by adammhaile** [comment]
Yeah, unfortunately, I can't think of one either. rrgit was really just a "best I could do with what I had" sort of thing. And the date checking stuff is really hacky, but it's what I've got. When you push to duet, it actually pushes, then checks what the remote date/time is, and then forces your local version to have the same time stamp.... like I said, hacky :) 
Well - I'm wrong, there is a solution, but I don't like it. That would be to always re-pull all files (even if only into memory) when checking for remote/local differences and actually check if the file contents are diff. 
Right now I just shortcut it with date and size.

**Comment by AndyEveritt** [comment]
How do you feel about attempting the full comparison method? 😉

**Comment by adammhaile** [comment]
I'm not completely opposed to it, but it comes with the issue that, depending on what files you have included/excluded, it could take a long time. For example, if you removed the default `*.bin` exclusion from your .rrgit config it would check all the firmware files every time - some of them are decently large and slow to download. If you were only syncing pure text files, most are small so it probably wouldn't be too bad.
I feel it's possible to do but it would have to be an option that's not on by default. Something that you could add to your .rrgit config after the fact and enable. Basically telling it "I'm using this on multiple machines, be smarter about your diff checking".

Ideally what'd we'd need is for @dc42 to add an RRF side file hash (or allow us to submit that change) so that we could simply ask RRF for the file hash without downloading the file first. Basically, what git does :)

**Comment by dc42** [comment]
Is M38 any use to you? https://duet3d.dozuki.com/Wiki/M38

**Comment by pfn** [comment]
> Is M38 any use to you? https://duet3d.dozuki.com/Wiki/M38

@dc42 this is a good suggestion, but the rr_gcode -> rr_reply mechanism isn't reliable. I tried spamming M38 rr_gcode/rr_reply and the response buffers got lost, or out of sync

This is an example of requesting M38 for all files and subsequently fetching rr_reply. If there were some key system that could fetch corresponding responses, it could be more reliable, but this isn't usable  :(
```
...
sys\resurrect-prologue.g = []
sys\sleep.g = [853ec55ec032eb57f94504639f944c6066c7fac0
4389d7c852c965f494f5591099ad6a115285e5e7
]
...
